File: 251112/숫자로 이루어진 사각형/rectangle-with-a-number.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const N = Number(input[0]);
// Please write your code here.
let num = 1;
let result = "";

for (let i = 0; i < N; i++) {
    let line = "";
    for (let j = 0; j < N; j++) {
        line += num + " ";
        num++;
        if (num === 10) num = 1; // 9 다음에는 다시 1
    }
    result += line.trim() + "\n";
}

console.log(result.trim());

File: 250326/숫자로 이루어진 사각형/rectangle-with-a-number.java
import java.util.Scanner;
public class Main {
    public static void printSquare(int n) {
        int a = 1;
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                if( (a) > 9) {
                    a = 1;
                }
                System.out.print(a + " ");
                a++;
                
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        // Please write your code here.
        printSquare(n);

    }
}